scripts/zkai_minimal_transformer_block_benchmark_gate.py

```
@@ -0,0 +1,666 @@
+            current = current / part
+            part_stat = current.lstat()
+            if stat.S_ISLNK(part_stat.st_mode):
+                raise MinimalBlockBenchmarkError(f"source path traverses symlink: {path}")
+            pre_stat = part_stat
+        if pre_stat is None or not stat.S_ISREG(pre_stat.st_mode):
+            raise MinimalBlockBenchmarkError(f"source path is not a regular file: {path}")
+        fd = os.open(candidate, os.O_RDONLY | getattr(os, "O_NOFOLLOW", 0))
+        try:
+            post_stat = os.fstat(fd)
+            if (post_stat.st_dev, post_stat.st_ino) != (pre_stat.st_dev, pre_stat.st_ino):
+                raise MinimalBlockBenchmarkError(f"source changed while reading: {path}")
```

Ruff (0.15.12)

[warning] Avoid specifying long messages outside the exception class

(TRY003)